packages/eslint-config-custom/index.js

```diff
@@ -35,6 +35,7 @@ module.exports = {
     '@typescript-eslint/no-confusing-void-expression': ['error', { ignoreArrowShorthand: true }],
     '@typescript-eslint/switch-exhaustiveness-check': 'error',
     '@typescript-eslint/no-unnecessary-condition': ['error', { allowConstantLoopConditions: true }],
+    '@typescript-eslint/no-invalid-void-type': 'off',
```

Comment:
This is weird — according to [the docs](https://typescript-eslint.io/rules/no-invalid-void-type/#allowingenerictypearguments), it shouldn't give a linting error when void is used in a generic type argument. Did you look into this? I just tried re-enabling the rule with `allowInGenericTypeArguments: true`, and it's still giving an error.

Not necessary to fix in this PR, but if you know the reason, it'd be great to not have to disable this rule.

Reply:
notice the docs are regarding type definitions, not function calls - in `Promise.withResolvers<void>()`, it's not a generic, you're parameterizing the types of actual values on the returned object